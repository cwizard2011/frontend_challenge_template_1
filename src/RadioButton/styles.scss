/* The radio-button-container */
.radio-button-container {
  display: block;
  position: relative;
  padding-left: 35px;
  margin-bottom: 12px;
  cursor: pointer;
  font-size: 20px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin-left: 15px;
}

.text-span {
  font-size: 12px;
}

/* Hide the browser's default radio button */
.radio-button-container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

/* Create a custom radio button */
.checkmark {
  position: absolute;
  top: 0;
  left: 0;
  height: 18px;
  width: 18px;
  background-color: #eee;
  border-radius: 50%;
  margin-top: 8px;
}

/* On mouse-over, add a grey background color */
.radio-button-container:hover input ~ .checkmark {
  background-color: #ccc;
}

/* When the radio button is checked, add a blue background */
.radio-button-container input:checked ~ .checkmark {
  background-color: #f7436b;
}

/* Create the indicator (the dot/circle - hidden when not checked) */
.checkmark:after {
  content: "";
  position: absolute;
  display: none;
}

/* Show the indicator (dot/circle) when checked */
.radio-button-container input:checked ~ .checkmark:after {
  display: block;
}

/* Style the indicator (dot/circle) */
.radio-button-container .checkmark:after {
 	top: 5px;
	left: 5px;
	width: 8px;
	height: 8px;
	border-radius: 50%;
	background: white;
}