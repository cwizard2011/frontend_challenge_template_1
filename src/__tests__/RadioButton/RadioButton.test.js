import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import RadioButton from '../../RadioButton';

describe('<RadioButton />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<RadioButton />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  })
})