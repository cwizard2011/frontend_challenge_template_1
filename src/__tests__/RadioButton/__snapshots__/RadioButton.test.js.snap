// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> renders without crashing 1`] = `
<label
  className="radio-button-container"
>
  <span
    className="text-span"
  />
  <input
    checked="checked"
    onClick={[Function]}
    type="radio"
  />
  <span
    className="checkmark"
  />
</label>
`;
